junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ExtensionRegistry.java

```diff
@@ -88,28 +90,47 @@ private ExtensionRegistry(ExtensionRegistry parent) {
 	}
 
 	/**
-	 * Get all {@code Extensions} in this registry or one of its ancestors of
+	 * Stream all {@code Extensions} in this registry or one of its ancestors of
 	 * the specified type.
 	 *
 	 * @param extensionType the type of {@link Extension} to get
 	 * @see #getReversedExtensions(Class)
-	 * @see #stream(Class)
+	 * @see #getExtensions(Class)
 	 */
-	public <E extends Extension> List<E> getExtensions(Class<E> extensionType) {
-		List<E> extensions = new ArrayList<>();
-
-		if (this.parent != null) {
-			extensions.addAll(this.parent.getExtensions(extensionType));
+	public <E extends Extension> Stream<E> streamExtensions(Class<E> extensionType) {
```

Comment:
> While I certainly appreciate your attention to detail, let's not go overboard here. This is just an internal API. :wink:

Haha, fair enough!

> Although I can live with streamExtensions, I will personally veto getExtensionsAsStream since there is no need to be overly verbose here.

Okay then. I can see that I was overly bikeshedding and letting myself be OCD here, so I'm happy for `streamExtensions` to be as-is. :smiley: